src/pages/Admin/Venue/Rooms/RoomsForm.tsx

```diff
@@ -40,10 +41,13 @@ export const RoomsForm: React.FC = () => {
 
   useEffect(() => {
     const fetchVenueFromAPI = async () => {
+      if (!venueId) return history.replace("/admin");
+
```

Comment:
Why are we redirecting within this function? I assume the intent of this addition was to ensure `venueId` isn't `undefined`/similar by the time it gets passed to `.doc()`. Why wouldn't we just `return` early like we do with all of our other guard patterns like this?

```suggestion
      if (!venueId) return
```

Reply:
I see that later on this same `history.replace` code is used, so that makes sense.

Though it seems we could move this guard outside of the `async` `fetchVenueFromAPI` function and just have it in the root of the `useEffect`?

Reply:
I think so, but it still would be the same

Reply:
Thats fine. I think it just more confused me being inside the inner function, rather than the root of the `useEffect`